src/python/pants/engine/isolated_process.py

```diff
@@ -108,8 +138,20 @@ def __init__(self, exit_code, stdout, stderr, process_description):
     super().__init__(msg)
 
 
-@rule(ExecuteProcessResult, [FallibleExecuteProcessResult, ExecuteProcessRequest])
-def fallible_to_exec_result_or_raise(fallible_result, request):
+@rule(ProductDescription, [ExecuteProcessRequest])
+def get_request_description(req):
+  return ProductDescription(req.description)
+
+
+@rule(ProductDescription, [MultiPlatformExecuteProcessRequest])
+def get_multi_platform_request_description(req):
+  # we can safely extract the first description because we guarantee that at
+  # least one request exists and that all of their descriptions are the same.
```

Comment:
Maybe move this into a helper method of `MultiPlatformExecuteProcessRequest` to help ensure that assumption?